### ISSUE TITLE
Bimviews loading forever...

### ISSUE THREAD

**Issue by mantinha** [opensourceBIM/BIMserver#1042]
I need to start Bimserver using the arguments for the plugin folders and the Bimviews plugin should load normally.

I don't want Bimserver to download plugins automatically because I want to customize Bimviews.

Bimviews reads which data to load and which data is in which class?

![plugins arguments](https://user-images.githubusercontent.com/25441376/66776683-567fc100-ee9d-11e9-845f-e76161ee5708.png)

![Bimviews loading](https://user-images.githubusercontent.com/25441376/66776547-04d73680-ee9d-11e9-94c4-91dc9d1d4587.png)

**Comment by rubendel** [comment]
Have you at least opened the JavaScript console? What's in the BIMserver logfile? Which browser is this?

**Comment by mantinha** [comment]
I tried again and added the Javascript API as a plugin, but some Bimsurfer feature was not loaded.

<img width="950" alt="bimsurfer_erro" src="https://user-images.githubusercontent.com/25441376/66786764-f9453900-eeb7-11e9-96f2-c4b22cc65479.png">

bimserver log file:
2019-10-14 19:17:47,122 INFO [main] o.b.BimServer [BimServer.java:245] Starting BIMserver (Windows 10, 1.8.0_221, 64bit, Xmx: 3,54 GB)
2019-10-14 19:17:49,329 INFO [main] o.b.BimServer [BimServer.java:333] Version: 1.5.168-SNAPSHOT
2019-10-14 19:17:49,329 INFO [main] o.b.BimServer [BimServer.java:342] Using "tmptestdata\home" as homedir
2019-10-14 19:17:51,043 INFO [pool-2-thread-1] o.b.e.PackageMetaData [PackageMetaData.java:77] Initializing IFC2X3TC1 with ifc2x3tc1
2019-10-14 19:17:51,056 INFO [pool-2-thread-2] o.b.e.PackageMetaData [PackageMetaData.java:77] Initializing IFC4 with ifc4
2019-10-14 19:17:51,067 INFO [pool-2-thread-3] o.b.e.PackageMetaData [PackageMetaData.java:77] Initializing GEOMETRY with geometry
2019-10-14 19:17:51,067 INFO [pool-2-thread-4] o.b.e.PackageMetaData [PackageMetaData.java:77] Initializing STORE with store
2019-10-14 19:17:51,068 INFO [pool-2-thread-3] o.b.e.PackageMetaData [PackageMetaData.java:77] Initializing LOG with log
2019-10-14 19:17:52,609 INFO [main] o.b.d.b.BerkeleyKeyValueStore [BerkeleyKeyValueStore.java:85] Non-empty database directory found "tmptestdata\home\database"
2019-10-14 19:17:53,738 INFO [main] o.b.BimServer [BimServer.java:811] Checking for stale records, this can take some time...
2019-10-14 19:17:53,770 INFO [main] o.b.BimServer [BimServer.java:841] Done checking for stale records (31 ms)
2019-10-14 19:17:54,169 INFO [main] o.b.ServerInfoManager [ServerInfoManager.java:115] Changing server state to RUNNING
2019-10-14 19:17:54,170 INFO [main] o.b.BimServer [BimServer.java:1107] Initializing database dependant logic...
2019-10-14 19:17:54,202 INFO [main] o.b.BimServer [BimServer.java:1258] Done initializing database dependant logic (32ms)
2019-10-14 19:17:55,227 INFO [main] o.b.LocalDevBimServerStarter [AbstractLocalDevBimServerStarter.java:91] Loading plugins...
2019-10-14 19:17:55,305 INFO [main] o.b.LocalDevBimServerStarter [AbstractLocalDevBimServerStarter.java:95] All plugins loaded (77 ms)

-------------------------------------------------
Google Chrome version 77.0.3865.90 64 bits

**Comment by rubendel** [comment]
Ah you need https://github.com/opensourceBIM/BIMsurfer-before2019, BIMvie.ws still uses that version and yet the new version of BIMsurfer.

**Comment by mantinha** [comment]
I loaded the server again and it looks like the error is in settings.js. Should I reference the Bimsurfer v2 and v3 libraries in Bimviews?

State:
![Bimsurfer status](https://user-images.githubusercontent.com/25441376/66831634-85924300-ef2e-11e9-8ffe-817241a6e538.png)
Console:
![Bimsurfer erro](https://user-images.githubusercontent.com/25441376/66831376-de151080-ef2d-11e9-9c1a-b2fd3877709b.png)

BIMserver log file:
2019-10-15 09:10:16,963 INFO [main] o.b.r.ClasspathResourceFetcher [ClasspathResourceFetcher.java:61] logback.xml sun.misc.Launcher$AppClassLoader@73d16e93
2019-10-15 09:10:16,968 INFO [main] o.b.r.ClasspathResourceFetcher [ClasspathResourceFetcher.java:62] Count: sun.misc.CompoundEnumeration@22a67b4
2019-10-15 09:10:16,976 INFO [main] o.b.BimServer [BimServer.java:248] Starting BIMserver (Windows 7, 1.8.0_111, 64bit, Xmx: 7,10 GB)
2019-10-15 09:10:18,178 INFO [main] o.b.BimServer [BimServer.java:251] SServiceMap Created
2019-10-15 09:10:18,408 INFO [main] o.b.BimServer [BimServer.java:254] JsonSocketReflectorFactory created
2019-10-15 09:10:18,813 INFO [main] o.b.BimServer [BimServer.java:257] ServerInfoManager created
2019-10-15 09:10:18,821 INFO [main] o.b.BimServer [BimServer.java:260] NotificationsManager created
2019-10-15 09:10:18,824 INFO [main] o.b.BimServer [BimServer.java:263] InternalServicesManager created
2019-10-15 09:10:18,826 INFO [main] o.b.BimServer [BimServer.java:266] PublicInterfaceFactory created
2019-10-15 09:10:19,124 INFO [main] o.b.r.ClasspathResourceFetcher [ClasspathResourceFetcher.java:61] pom.xml sun.misc.Launcher$AppClassLoader@73d16e93
2019-10-15 09:10:19,124 INFO [main] o.b.r.ClasspathResourceFetcher [ClasspathResourceFetcher.java:62] Count: sun.misc.CompoundEnumeration@5398edd0
2019-10-15 09:10:19,142 INFO [main] o.b.BimServer [BimServer.java:295] Embedded webserver created
2019-10-15 09:10:19,142 INFO [main] o.b.BimServer [BimServer.java:298] Done with initial setup
2019-10-15 09:10:19,376 INFO [main] o.b.EmbeddedWebServer [EmbeddedWebServer.java:88] Modo Desenvolvedor: VERIFICADO
2019-10-15 09:10:19,388 INFO [main] o.b.BimServer [BimServer.java:336] Iniciando BIMserver: Version: 1.5.174-SNAPSHOT
2019-10-15 09:10:19,388 INFO [main] o.b.BimServer [BimServer.java:345] Usando "C:\Test" como diretorio principal
2019-10-15 09:10:19,390 INFO [main] o.b.e.MetaDataManager [MetaDataManager.java:81] Metadados: nop: java.io.PrintStream@56f2bbea
2019-10-15 09:10:21,324 INFO [pool-2-thread-2] o.b.e.PackageMetaData [PackageMetaData.java:77] Initializing IFC4 with ifc4
2019-10-15 09:10:21,324 INFO [pool-2-thread-1] o.b.e.PackageMetaData [PackageMetaData.java:77] Initializing IFC2X3TC1 with ifc2x3tc1
2019-10-15 09:10:21,324 INFO [pool-2-thread-4] o.b.e.PackageMetaData [PackageMetaData.java:77] Initializing STORE with store
2019-10-15 09:10:21,324 INFO [pool-2-thread-3] o.b.e.PackageMetaData [PackageMetaData.java:77] Initializing GEOMETRY with geometry
2019-10-15 09:10:21,324 INFO [pool-2-thread-5] o.b.e.PackageMetaData [PackageMetaData.java:77] Initializing LOG with log
2019-10-15 09:10:22,829 INFO [main] o.b.BimServer [BimServer.java:674] No email templates found
2019-10-15 09:10:22,837 INFO [main] o.b.d.b.BerkeleyKeyValueStore [BerkeleyKeyValueStore.java:87] Non-empty database directory found "C:\Test\database"
2019-10-15 09:10:23,462 INFO [main] o.b.d.Database [Database.java:106] Carregando pacotes: store
2019-10-15 09:10:23,463 INFO [main] o.b.d.Database [Database.java:108] Carregando pacotes: log
2019-10-15 09:10:23,463 INFO [main] o.b.d.Database [Database.java:110] Carregando pacotes: geometry
2019-10-15 09:10:23,492 INFO [main] o.b.d.Database [Database.java:156] Registro no bd: CONSULTADO :7d60d593-e773-400f-a683-bf5d45c9a508
2019-10-15 09:10:23,991 INFO [main] o.b.BimServer [BimServer.java:729] Encriptando dados..
2019-10-15 09:10:23,992 INFO [main] o.b.BimServer [BimServer.java:822] Checking for stale records, this can take some time...
2019-10-15 09:10:23,993 INFO [main] o.b.BimServer [BimServer.java:825] Checando projetos..
2019-10-15 09:10:24,037 INFO [main] o.b.BimServer [BimServer.java:853] Done checking for stale records (43 ms)
2019-10-15 09:10:24,505 INFO [main] o.b.ServerInfoManager [ServerInfoManager.java:115] Changing server state to RUNNING
2019-10-15 09:10:24,506 INFO [main] o.b.BimServer [BimServer.java:1123] Initializing database dependant logic...
2019-10-15 09:10:26,741 INFO [main] o.i.IfcOpenShellEnginePlugin [IfcOpenShellEnginePlugin.java:101] Using overruled system setting for commit sha
2019-10-15 09:10:26,755 INFO [main] o.i.IfcOpenShellEnginePlugin [IfcOpenShellEnginePlugin.java:109] Using C:\Test\tmp\org.opensourcebim.ifcopenshellplugin\IfcGeomServer-v0.6.0-b4ce5be-win64.exe
2019-10-15 09:10:26,795 INFO [main] o.i.IfcOpenShellEnginePlugin [IfcOpenShellEnginePlugin.java:101] Using overruled system setting for commit sha
2019-10-15 09:10:26,796 INFO [main] o.i.IfcOpenShellEnginePlugin [IfcOpenShellEnginePlugin.java:109] Using C:\Test\tmp\org.opensourcebim.ifcopenshellplugin\IfcGeomServer-v0.6.0-b4ce5be-win64.exe
2019-10-15 09:10:32,373 INFO [main] o.b.BimServer [BimServer.java:1291] Done initializing database dependant logic (7866ms)
2019-10-15 09:10:33,105 INFO [main] o.b.LocalDevBimServerStarter [AbstractLocalDevBimServerStarter.java:103] ServerState = RUNNING
2019-10-15 09:10:33,105 INFO [main] o.b.LocalDevBimServerStarter [AbstractLocalDevBimServerStarter.java:105] Loading plugins...
2019-10-15 09:10:33,375 INFO [main] o.b.LocalDevBimServerStarter [AbstractLocalDevBimServerStarter.java:109] All plugins loaded (269 ms)

------------------------------------------------------------------------
Google Chrome Version 77.0.3865.90 64 bits

**Comment by mantinha** [comment]
Is LazyLoad code right? Part of him seems to be wrong. The way it was written is not easy to understand.

![image](https://user-images.githubusercontent.com/25441376/66837448-6351f280-ef39-11e9-856f-f1537c01c819.png)

**Comment by rubendel** [comment]
Sorry it's a bit of a mess, you also need to use the V1 branch on that repository

**Comment by mantinha** [comment]
I tried again with version 1 of Bimsurfer but the error happened again. I didn't find the "SceneJS.configure" method I don't know where it comes from

![image](https://user-images.githubusercontent.com/25441376/66859198-36193a80-ef61-11e9-9c94-dc59618f36a6.png)
---------------------------------------------------------------------
Google Chrome Version 77.0.3865.90 64 bits

**Comment by rubendel** [comment]
The errors are the result of the 404's, what's weird is that the URL contains uppercase characters, when I run BIMvie.ws locally BIMsurfer V1 is at `http://localhost:8080/apps/bimsurfer`. Did you change anything?

**Comment by mantinha** [comment]
I didn't look at that detail. Thank you for opening my eyes.

Between my tests I had modified the getBimSurferApiAdress method return to "../bimsurfer" to "../BIMsurfer"

Now I fixed it and bimviews was executed. I can proceed to the next step. "Administrator Username and Password"

I am closing this topic. I will return soon ..
![image](https://user-images.githubusercontent.com/25441376/66862018-74fdbf00-ef66-11e9-8880-9bedc53ef189.png)
